### ISSUE TITLE
# **Fraudulent Contract Analysis**

### ISSUE THREAD

**Issue by BankkRoll** [ApebottingETH/MevBot#9]
# **Fraudulent Contract Analysis**
![Alternative Text](https://bestanimations.com/media/hazards/2108418205warning-red-blinking-sign-animated-gif-4.gif)

I've analyzed the provided code and it appears to be a fraudulent contract. It claims to be an MEV bot, which should allow users to exploit inefficiencies in the market and get profits. However, the actual functionality of the contract is limited to transferring the Ether balance of the contract to another address.

The key part of the code is the `Start()` function:

```solidity
function Start() public payable {
    emit Log("Running MEV action. This can take a while; please wait..");
    payable(_callMEVAction()).transfer(address(this).balance);
}
```
The `_callMEVAction()` function does not perform any actual MEV-related actions but instead returns an address parsed from a hardcoded string. The Start() function transfers the entire balance of the contract to this address.

Additionally, the code contains several functions that seem to be there just to make it look more complicated and legitimate, like `memcpy`, `orderContractsByLiquidity`, `calcLiquidityInContract`, and others. However, these functions are not used in any meaningful way in the contract.

In conclusion, this contract appears to be a scam, and I would not recommend using it. It's essential to be cautious and critical when dealing with smart contracts and to ensure that the code comes from a trusted source.
